frontend/pages/hosts/details/cards/HostSummary/MacSettingsIndicator/MacSettingsIndicator.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import ReactTooltip from "react-tooltip";
+import Icon from "components/Icon";
+import Button from "components/buttons/Button";
+import { IMacSettings, MacSettingsStatus } from "interfaces/mdm";
+
+const baseClass = "mac-settings-indicator";
+
+interface IMacSettingsIndicatorProps {
+  profiles: IMacSettings;
+  onClick?: () => void;
+}
+const MacSettingsIndicator = ({
+  profiles,
+  onClick,
+}: IMacSettingsIndicatorProps): JSX.Element => {
+  const STATUS_DISPLAY_OPTIONS = {
+    Latest: {
+      iconName: "success",
+      tooltipText: "Host applied the latest settings",
+    },
+    Pending: {
+      iconName: "pending",
+      tooltipText: "Host will apply the latest settings when it comes online",
+    },
+    Failing: {
+      iconName: "error",
+      tooltipText:
+        "Host failed to apply the latest settings. Click to view error(s).",
+    },
+  } as const;
+
+  const getMacSettingsStatus = (
+    hostMacSettings: IMacSettings | undefined
+  ): MacSettingsStatus => {
+    const statuses = hostMacSettings?.map((setting) => setting.status);
+    if (statuses?.includes("failed")) {
+      return "Failing";
+    }
+    if (statuses?.includes("pending")) {
+      return "Pending";
+    }
+    return "Latest";
```

Comment:
Note: This will return "Latest" if `statuses` is `null` or `undefined` or an empty array. This isn't a problem currently since the `MacSettingsIndicator` is being conditionally rendered within the `HostSummary` card and there is a falsey check happening there (which would catch the `null` and `undefined` but not an empty array, FWIW). I don't think we need to delay this PR necessarily, but wanted to call it out here for future reference since this component might get re-used in another context and this behavior might not be expected.

Reply:
Thanks @gillespi314, good point